kernel/aster-nix/src/device/mod.rs

```diff
@@ -3,17 +3,17 @@
 mod null;
 mod pty;
 mod random;
-#[cfg(feature = "intel_tdx")]
+#[cfg(all(target_arch = "x86_64", feature = "intel_tdx"))]
 mod tdxguest;
 pub mod tty;
 mod urandom;
 mod zero;
 
 pub use pty::{new_pty_pair, PtyMaster, PtySlave};
 pub use random::Random;
-#[cfg(feature = "intel_tdx")]
+#[cfg(all(target_arch = "x86_64", feature = "intel_tdx"))]
 use tdx_guest::tdx_is_enabled;
-#[cfg(feature = "intel_tdx")]
+#[cfg(all(target_arch = "x86_64", feature = "intel_tdx"))]
 pub use tdxguest::TdxGuest;
```

Comment:
It is a pain-in-the-ass to write and read long cfg like `#[cfg(all(target_arch = "x86_64", feature = "intel_tdx"))]`.

So instead of having three separate, long cfg-gated code blocks like

```rust
#[cfg(all(target_arch = "x86_64", feature = "intel_tdx"))]
mod tdxguest;

#[cfg(all(target_arch = "x86_64", feature = "intel_tdx"))]
use tdx_guest::tdx_is_enabled;

#[cfg(all(target_arch = "x86_64", feature = "intel_tdx"))]
pub use tdxguest::TdxGuest;
```

we can use `cfg_if` to group them into one.

```rust
cfg_if! {
    if #[cfg(all(target_arch = "x86_64", feature = "intel_tdx"))] {
        mod tdxguest;

        use tdx_guest::tdx_is_enabled;

        pub use tdxguest::TdxGuest;
    }
}
```

Use the same technique to improve the readability of other TDX-related modules.